pkg/server/endpoints/registration/handler.go

```diff
@@ -486,8 +563,9 @@ func (h *Handler) CreateJoinToken(
 		},
 	})
 	if err != nil {
+		err = status.Error(codes.Internal, "Failed to register token")
```

Comment:
We should be embellishing the error with the status code after it is logged. In this particular instance, we were logging the error and returning an (intentionally?) vague error description. With this change we no longer log the real reason for the failure.

Reply:
My mistake, meant to wrap the existing error.

However, I think this is more of an argument that datastore `CreateJoinToken` (and rest of API) should return actual codes, and document that it does so, so that callers can just return the straight error knowing a code is embedded (better yet, why not be returning `Status` objects?).
Generally `Internal` would be the way to go for some DB failure, though it is possible the failure in this case is `AlreadyExists` if the token already exists.